pkg/apis/sources/v1alpha1/ping_conversion.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2020 The Knative Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"context"
+	"fmt"
+	"knative.dev/eventing/pkg/apis/sources/v1alpha2"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"reflect"
+)
+
+// ConvertUp implements apis.Convertible.
+// Converts source (from v1alpha1.PingSource) into v1alpha2.PingSource
+func (source *PingSource) ConvertUp(ctx context.Context, obj apis.Convertible) error {
+	switch sink := obj.(type) {
+	case *v1alpha2.PingSource:
+		sink.ObjectMeta = source.ObjectMeta
+		sink.Spec = v1alpha2.PingSourceSpec{
+			Schedule: source.Spec.Schedule,
+			JsonData: source.Spec.Data,
+		}
+		sink.Status = v1alpha2.PingSourceStatus{
+			SourceStatus: duckv1.SourceStatus{
+				Status:  source.Status.Status,
+				SinkURI: source.Status.SinkURI,
+			},
+		}
+		// Optionals
+		if source.Spec.Sink != nil {
+			sink.Spec.Sink = *source.Spec.Sink.DeepCopy()
```

Comment:
No need to DeepCopy().

Reply:
I modify it, so I don't want to make mistakes

Reply:
I don't see where. AFAIK, the conversion framework do not mutate source, or obj, just marshaling obj. I'm just trying to understand because I didn't do this in the flows conversion webhooks and that might be wrong.

Reply:
@dprotaso ?

Reply:
I'd view this from the perspective of the `Convertible` interface and not look at the conversion webhook behaviour.

If I were to convert `x.v1` to say `x.v2` subsequent modifications to `x.v1` shouldn't have a side-effect of modifying the new instance of `x.v2`

Reply:
Note - I don't think the APIs in `serving` follow this strictly but they should. 

I say that because technically these are public APIs that can be consumed outside of the conversion webhook.

Reply:
Also @lionelvillard is correct wrt. to the conversion webhook framework - you don't need to worry about side-effects since it's just a doing a transformation.

Secondly, at least for serving controllers, `DeepCopy` the informer copy before it tried to `ConvertUp`

Reply:
ok I am going to keep playing it safe for things that are pointers.